In [1]:
import numpy as np

# Size of the structure 
let $a$ and $b$ are the major and minor diameter directly measered from ALLADIN in unit of degree. First of all, the angular size is converted into radian using the equation given below; 
- $  180^o=\pi^c$
- $ a^c = \frac{\pi^c}{180^o}a^o $
- $ b^c = \frac{\pi^c}{180^o}b^o $\
The code to convert angle in radian is given below; 

In [2]:
a_deg = 23.59/60  # initially it was in minute, converted into degree dividing by 60
b_deg = 13.8/60
a_rad = np.pi*a_deg/180
b_rad = np.pi*b_deg/180

In [3]:
a_rad, b_rad

(0.006862052842424372, 0.004014257279586958)

# If D be the distance of the nebula /or cavity from us (earth) then the angular size of major and moinor diameter are converted into linear size using the formula; 
#### - $a_{linear} = D \times a_{rad}$
#### - $b_{linear} = D \times b_{rad}$


#### First define distance D

In [4]:
D = 309.98 # in pc, this is different for different structure 


In [5]:
a_linear = D*a_rad
b_linear = D*b_rad

In [6]:
a_linear, b_linear

(2.127099140094707, 1.2443394715263654)

### Thus, the size of the structure is, $a_{linear}\times b_{linear} = 2.13 pc \times 1.24 pc$

### Calculate mean diameter and radius 

## The mean diameter is given as
## $d_{mean} = \frac{a_{linear} + b_{linear}}{2}$

In [7]:
d_mean = (a_linear + b_linear)/2

In [8]:
d_mean

1.6857193058105362

## Thus, the mean diameter is 1.69 pc

## The mean radius is given as,
###  - $R_{mean}= \frac{d_{mean}}{2}$

In [9]:
R_mean = d_mean/2
R_mean, R_mean*3.085e16

(0.8428596529052681, 2.600222029212752e+16)

### Thus, the average radius of structure is 0. pc

# Jeans Criteria : Densiy and Jeans Mass

### The density of the structure is calculated as;
## - $\rho = (\frac{3}{4\pi})^{2/3} \frac{k_b T}{m_HGR^2}$

In [14]:
T = 28.68 # K
m_H = 1.67e-27 
R = R_mean*3.0857e16 
k_b = 1.38e-23
G = 6.67e-11

In [15]:
density = (3/(4*np.pi))**(2/3)*k_b*T/(m_H*G*R**2)
density

2.021493074689246e-18

### The Jeans Mass of the structure is calculated as;
## - $M_J =  (\frac{k_b T}{m_HG})^{3/2}\frac{1}{\rho^{1/2}} $

In [16]:
M_J = (k_b*T/(m_H*G))**(3/2)*1/np.sqrt(density)
M_J

1.4896620828259788e+32

In [17]:
M_J_Sun = M_J/(1.989e+30)
M_J_Sun

74.89502678863644